src/Nethermind/Nethermind.Network/PeerManager.cs

```diff
@@ -399,6 +409,31 @@ private bool EnsureAvailableActivePeerSlot()
             return AvailableActivePeersCount - _pending > 0;
         }
 
+        private bool CanMakeRoomForHigherDiversityPeer()
```

Comment:
Make it accept a parameter, the ID or something, change it to `CanReplaceActivePeer`.

Reply:
Therefore, dont call it in `EnsureAvailableActivePeerSlot`.